test/sslapitest.c

```diff
@@ -5030,12 +5037,63 @@ static int test_key_exchange(int idx)
             kexch_alg = NID_ffdhe8192;
             kexch_name0 = "ffdhe8192";
             break;
+# endif
+# ifndef OPENSSL_NO_ML_KEM
+#  if !defined(OPENSSL_NO_TLS1_2)
+        case 16:
+            max_version = TLS1_2_VERSION;
+#   if !defined(OPENSSL_NO_EC)
+            /* Set at least one EC group so the handshake completes */
+            kexch_names = "MLKEM512:MLKEM768:MLKEM1024:secp256r1";
+#   elif !defined(OPENSSL_NO_DH)
+            kexch_names = "MLKEM512:MLKEM768:MLKEM1024";
+#   else
+            /* With neither EC nor DH TLS 1.2 can't happen */
+            return 1;
+#   endif
+#  endif
+            /* Fall through */
+        case 12:
+            kexch_groups = NULL;
```

Comment:
So, we're not testing setting by id here? Note you *don't* need a nid for this. You just need to know the group id and or it with `TLSEXT_nid_unknown` See the documentation for `SSL_set1_groups`.

Reply:
Sadly, that's not enough, because the `TLSEXT_nid_unknown` only works in *get* methods, I did not find any interfaces for setting  a numeric group list using `TLSEXT_nid_unknown|codepoint`.